lib/jnpr/junos/transport/tty_serial.py

```diff
@@ -91,6 +91,9 @@ def read_prompt(self):
                 break         # done reading
         else:
             # exceeded the while loop timeout
+            self._ser.reset_input_buffer()
             return (None, None)
 
+        # clear the input buffer, the next read will refill as necessary
+        self._ser.reset_input_buffer()
```

Comment:
@nembery @vnitinv 

This change (and/or the one on line 94) seems to have cause this ERROR message in unit tests:
```
======================================================================
ERROR: test_console_serial (tests.unit.test_console.TestConsole)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/virtualenv/python2.7.9/lib/python2.7/site-packages/mock.py", line 1201, in patched
    return func(*args, **keywargs)
  File "/home/travis/build/Juniper/py-junos-eznc/tests/unit/test_console.py", line 153, in test_console_serial
    self.dev.open()
  File "/home/travis/build/Juniper/py-junos-eznc/lib/jnpr/junos/console.py", line 190, in open
    raise ex
SerialException: Attempting to use a port that is not open
-------------------- >> begin captured logging << --------------------
jnpr.junos.tty: INFO: TTY: connecting to TTY:1.1.1.1:23 ...
jnpr.junos.tty: INFO: TTY: logging in......
jnpr.junos.tty_telnet: DEBUG: Got: 
 ogin:
jnpr.junos.tty_telnet: DEBUG: Got: 
 password:
jnpr.junos.tty_telnet: DEBUG: Got: 
root@device:~ # 
jnpr.junos.tty: INFO: TTY: OK.....starting NETCONF
jnpr.junos.tty: INFO: TTY: connecting to TTY:23 ...
jnpr.junos.tty: INFO: TTY: logging in......
jnpr.junos.console: ERROR: Exception occurred: login:Attempting to use a port that is not open
--------------------- >> end captured logging << ---------------------
======================================================================
```

Can one of you investigate that further?